src/main/java/com/commercetools/sync/products/helpers/VariantReferenceResolver.java

```diff
@@ -73,32 +77,48 @@ public VariantReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
     public CompletionStage<ProductVariantDraft> resolveReferences(
         @Nonnull final ProductVariantDraft productVariantDraft) {
         return resolvePricesReferences(ProductVariantDraftBuilder.of(productVariantDraft))
+            .thenCompose(this::resolveAssetsReferences)
             .thenCompose(this::resolveAttributesReferences)
             .thenApply(ProductVariantDraftBuilder::build);
     }
 
+    @Nonnull
+    CompletionStage<ProductVariantDraftBuilder> resolveAssetsReferences(
+        @Nonnull final ProductVariantDraftBuilder productVariantDraftBuilder) {
+        final List<AssetDraft> productVariantDraftAssets = productVariantDraftBuilder.getAssets();
+        if (productVariantDraftAssets == null) {
+            return completedFuture(productVariantDraftBuilder);
+        }
+
+        return mapValuesListToFutureOfCompletedValues(productVariantDraftAssets,
+            assetReferenceResolver::resolveReferences).thenApply(productVariantDraftBuilder::assets);
+    }
```

Comment:
It could be achieved by having an interface that both resolvers could implement:

````java
public interface AssetContainerResolver<T>{

    @Nonnull
    CompletionStage<T> resolveAssetsReferences(@Nonnull final T draftBuilder);

    @Nonnull
    default CompletionStage<T> resolveAssetsReferences(
        @Nonnull final T draftBuilder,
        @Nonnull final Function<T, List<AssetDraft>> assetsGetter,
        @Nonnull final BiFunction<T, List<AssetDraft>, T> assetsSetter,
        @Nonnull final AssetReferenceResolver assetReferenceResolver) {

        final List<AssetDraft> draftAssets = assetsGetter.apply(draftBuilder);
        if (draftAssets == null) {
            return completedFuture(draftBuilder);
        }

        return mapValuesListToFutureOfCompletedValues(draftAssets, assetReferenceResolver::resolveReferences)
            .thenApply(resolvedAssets -> assetsSetter.apply(draftBuilder, resolvedAssets));
    }
}
````

but then the implementing method of resolvers would have to have `resolveAssetReferences` as public which is something I don't want (at least for now) since resolvers by convention now have only `resolveReferences` method exposed and no other methods. 

Since in this case, there is not much improvement by reuse. I would wait if in the future for example other resources have assets then we could start thinking about exposing more methods of the resolvers and then I could reuse a common asset resolution interface.

Reply:
Why only by the interface? Both classes extend `BaseReferenceResolver` - the code could be implemented there.

Reply:
Not so easy, because in the BaseReferenceResolver the generic type is a *`Draft` not a *`DraftBuilder`. So might need a bit more refactoring. So I wouldn't do it in the same PR.

Reply:
ok